torch/testing/_internal/common_methods_invocations.py

```diff
@@ -500,6 +500,48 @@ def sample_inputs(self, device, dtype, requires_grad=False):
             ]
 
 
+class TriangularOpInfo(OpInfo):
+    """Operator information for function that take lower or upper triangular matrices as input.
+    They require a modified function to be tested for gradcheck, because the finite-difference algorithm
+    for calculating derivatives does not preserve the triangular property of the input and returning incorrect results.
+    """
+
+    def get_op(self):
+        """
+        Returns the function variant of the operator, torch.<op_name>,
+        compatible with gradcheck for triangular input functions.
+        It works only for single input argument and upper kwarg
+        """
+        def triangular_func(non_triangular_input, upper=False):
```

Comment:
This is clever, but instead of this approach should we just implement an input pre-processing function for gradcheck?

Reply:
(Possibly in a separate PR we put in before this while we investigate the performance issues?)

Reply:
I think we indeed should add the preprocessing function. I submitted an issue for this #50837.